frontend/src/pages/surveys/components/step/render-steps.tsx

```diff
@@ -0,0 +1,172 @@
+import { useAppDispatch, useCallback } from '#libs/hooks/hooks.js';
+import { type StepsType } from '#packages/survey/libs/types/types.js';
+import { SurveySteps } from '#packages/survey/survey.js';
+import {
+  FEELINGS_CATEGORIES,
+  GOALS_CATEGORIES,
+  JOURNALING_EXPERIENCE_CATEGORIES,
+  MEDITATION_EXPERIENCE_CATEGORIES,
+  PREFERENCES_CATEGORIES,
+  WORRIES_CATEGORIES,
+} from '#pages/surveys/libs/constants/constants.js';
+import { actions } from '#slices/survey/survey.js';
+
+import { Step } from './step.js';
+
+type StepsProperties = {
+  currentStep: StepsType;
+  onNextStep: () => void;
+  onPreviousStep: () => void;
+  onSubmit: () => void;
+};
+
+const RenderSteps: React.FC<StepsProperties> = ({
+  currentStep,
+  onNextStep,
+  onPreviousStep,
+  onSubmit,
+}): React.ReactNode => {
+  const dispatch = useAppDispatch();
+
+  const setPreference = useCallback(
+    (preferences: string[] | string): void => {
+      if (Array.isArray(preferences)) {
+        dispatch(actions.setPreferences(preferences));
+      }
+    },
+    [dispatch],
+  );
```

Comment:
I'm not sure about the complicity, but you can create one useCallback function instead of all of them, where you can pass dispatched action as input callback and boolean like `shouldBeArray`, that will check if string otherwise

Reply:
done!